core/src/env.rs

```diff
@@ -19,3 +23,74 @@ pub trait Sound {
 pub trait Input {
 
 }
+
+pub struct PixelPos {
```

Comment:
I would move this (as well as `PixelColor`) to `primitives.rs`. In my opinion the content of `env.rs` is dedicated for things that are related to "frontend" specific stuff which (for example) needs to be implemented for all "frontends" (e.g. desktop or web).